tests/mu_convert_enum_test.c

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <string.h>
+#include "test_helper.h"
+
+#include "azure_macro_utils/macro_utils.h"
+
+#include "mu_convert_enum_test.h"
+
+// This lists the values in the first enum
+#define TEST_CONVERT_FROM_ENUM_VALUES \
+    test_from_a, \
+    test_from_b, \
+    test_from_c
+
+#define TEST_CONVERT_TO_ENUM_VALUES \
+    test_to_a, \
+    test_to_b
+
+MU_DEFINE_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_FROM_ENUM_VALUES);
+MU_DEFINE_ENUM(TEST_CONVERT_TO_ENUM, TEST_CONVERT_TO_ENUM_VALUES);
+
+/* The following code should not compile :
+MU_DEFINE_CONVERT_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM,
+    test_from_a, test_to_a,
+    test_from_c, test_to_b);*/
+
+MU_DEFINE_CONVERT_ENUM_WITH_VALIDATION(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM,
+    test_from_a, test_to_a,
+    test_from_b, test_to_b,
+    test_from_c, test_to_b);
```

Comment:
The following code also compiles and passes all tests (windows, did not try gcc/clang)

```c
MU_DEFINE_CONVERT_ENUM_WITH_VALIDATION(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM,
    test_to_a, test_to_a,
    test_from_b, test_to_b,
    test_from_c, test_to_b);
```

Would be nice if the code would detect that "test_to_a" is not a value of "TEST_CONVERT_FROM_ENUM". But maybe that is something for the compiler to do or do not. There's no try.

Reply:
That would be cool if possible (I did something similar at runtime but I don't know if it can be written to fail at compile time)

---
In reply to: [394386497](https://github.com/Azure/azure-macro-utils-c/pull/36#discussion_r394386497) [](ancestors = 394386497)

Reply:
Done. At compile time with typedefes

---
In reply to: [394484222](https://github.com/Azure/azure-macro-utils-c/pull/36#discussion_r394484222) [](ancestors = 394484222,394386497)